src/Language/PureScript/Docs/AsHtml.hs

```diff
@@ -215,6 +215,35 @@ codeAsHtml r = outputWith elemAsHtml
             linkToDecl ns target mn (withClass class_ (text name))
         NoLink ->
           text name
+    Role role ->
+      case role of
+        "nominal" -> renderRole describeNominal "decl__role_nominal"
+        "phantom" -> renderRole describePhantom "decl__role_phantom"
+
+        -- representational is intentionally not rendered
+        "representational" -> toHtml ("" :: Text)
+
+        x -> P.internalError $ "codeAsHtml: unknown value for role annotation: '" <> T.unpack x <> "'"
+      where
+        renderRole hoverTextContent className =
+          H.a ! A.href (v docRepoRolePage) ! A.target (v "_blank") ! A.class_ "decl__role" $ do
+            H.abbr ! A.class_ "decl__role_hover" ! A.title (v hoverTextContent) $ do
+              H.sub ! A.class_ className $ do
+                toHtml ("" :: Text)
+
+        docRepoRolePage =
+          "https://github.com/purescript/documentation/blob/master/language/Roles.md"
+
+        describeNominal = fold
+          [ "The 'nominal' role means the type represented by this type "
+          , "parameter CANNOT be safely coerced to another type when one "
+          , "coerces the overall type to another type."
+          ]
+        describePhantom = fold
+          [ "The 'phantom' role means the type represented by this type "
+          , "parameter CAN be safely coerced to any other type when one "
+          , "coerces the overall type to another type."
+          ]
```

Comment:
(It's the argument that changes though, not the parameter!)

Reply:
Perhaps "this argument as represented by the type parameter..."? Though at the end of the day, this exact wording might not matter since the Pursuit docs will clarify the kind signatures further.